### PR TITLE
Try to clean up the workaround for the window resizing bug on Windows

### PR DESCRIPTION
**Pull request by tronical**
Use a dedicated cell and clean up the comments.

### CODE REVIEW

File: internal/backends/winit/glwindow.rs [hunk 1]
```diff
@@ -227,15 +229,19 @@ impl<Renderer: WinitCompatibleRenderer + 'static> WinitWindow for GLWindow<Rende
...
-        }
+        // slint::Window::set_size will call set_size() on this type, which would call
+        // set_inner_size on the winit Window. On Windows that triggers an new resize event
+        // in the next event loop iteration for mysterious reasons, with slightly different sizes.
+        // I suspect a bug in the way the frame decorations are added and subtracted from the size
+        // we provide.
+        // Work around it with this guard that prevents us from calling set_inner_size again.
+        assert!(!self.in_resize_event.get());
+        self.in_resize_event.set(true);
+        scopeguard::defer! { self.in_resize_event.set(false); }
```
Comment (ogoffart) [inline internal/backends/winit/glwindow.rs:240]: this is a bit weak to introduce a dependence just for that.  but that's a light dependency.

File: internal/backends/winit/glwindow.rs [hunk 2]
```diff
@@ -725,17 +731,15 @@ impl<Renderer: WinitCompatibleRenderer + 'static> WindowAdapterSealed for GLWind
     }
 
     fn set_size(&self, size: corelib::api::WindowSize) {
+        if self.in_resize_event.get() {
+            return;
+        }
         if let Ok(mut map_state) = self.map_state.try_borrow_mut() {
```
Comment (ogoffart) [inline internal/backends/winit/glwindow.rs:737]: so we don't need the _mut here anymore?  (that's what i read from the comment bellow)

File: internal/backends/winit/glwindow.rs (hunk 1 above)
Comment (tronical) [reply internal/backends/winit/glwindow.rs:240]: I can also do it without - no strong preference. Let me know what you prefer.

File: internal/backends/winit/glwindow.rs (hunk 2 above)
Comment (tronical) [reply internal/backends/winit/glwindow.rs:737]: The mut is still need for the assignment in line 740, I think. The comment was outdated.

File: internal/backends/winit/glwindow.rs (hunk 1 above)
Comment (ogoffart) [reply internal/backends/winit/glwindow.rs:240]: I think i prefer without so we don't add too many deps - no strong preference

**Review by ogoffart — APPROVED** [review]

File: internal/backends/winit/glwindow.rs (hunk 1 above)
Comment (tronical) [reply internal/backends/winit/glwindow.rs:240]: As a side-note: The reason why I used scopeguard is because it's already a dependency of corelib.

Comment (ogoffart) [reply internal/backends/winit/glwindow.rs:240]: keep it then

**Pull request merged**